src/lib.rs

```diff
@@ -40,81 +60,162 @@ impl<T: AsRef<[u8]>> Display for HexFmt<T> {
 
 impl<T: AsRef<[u8]>> LowerHex for HexFmt<T> {
     fn fmt(&self, f: &mut Formatter) -> Result {
+        Lowercase::fmt(self.0.as_ref(), f)
+    }
+}
+
+impl<T: AsRef<[u8]>> UpperHex for HexFmt<T> {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        Uppercase::fmt(self.0.as_ref(), f)
+    }
+}
+
+/// Wrapper for a list of byte arrays, whose `Debug`, `Display` and `LowerHex` implementations
+/// output shortened hexadecimal strings.
+pub struct HexList<T>(pub T);
+
+impl<T: Clone + IntoIterator> Debug for HexList<T>
+where
+    T::Item: AsRef<[u8]>,
```

Comment:
I am not a fan of mixing `where` clauses and inline trait requirements, they're so easy to miss this way. Maybe go 100% `where`?